pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/column/PhysicalColumnIndexContainer.java

```diff
@@ -54,11 +56,18 @@ public PhysicalColumnIndexContainer(SegmentDirectory.Reader segmentReader, Colum
     _readersByIndex = new HashMap<>();
     for (IndexType<?, ?, ?> indexType : IndexService.getInstance().getAllIndexes()) {
       if (segmentReader.hasIndexFor(columnName, indexType)) {
+        Map<String, Map<String, String>> columnProperties = indexLoadingConfig.getColumnProperties();
         IndexReaderFactory<?> readerProvider = indexType.getReaderFactory();
         try {
-          IndexReader reader = readerProvider.createIndexReader(segmentReader, fieldIndexConfigs, metadata);
-          if (reader != null) {
-            _readersByIndex.put(indexType, reader);
+          if (!indexType.getId().equals(StandardIndexes.TEXT_ID)
+                  || IndexLoadingConfig.processExistingSegments(columnName, columnProperties)) {
```

Comment:
I see one difference in conditions checked for text index between now vs earlier. 

Earlier (https://github.com/hypertrace/incubator-pinot/pull/317/files) the logic was: 
- first check if `IndexLoadingConfig.processExistingSegments` returns true. If yes, then create index.
- else check if `segmentReader.hasIndexFor()` returns true. If yes, then create index.

Now, the logic is:
- if `segmentReader.hasIndexFor()` returns true AND `IndexLoadingConfig.processExistingSegments()` returns true, then create index.